src/prefect/storage/bitbucket.py

```diff
@@ -80,14 +94,85 @@ def get_flow(self, flow_name: str) -> "Flow":
         Returns:
             - Flow: the requested flow
         """
+
+        return (
+            self._get_flow_cloud(flow_name)
+            if self.workspace
+            else self._get_flow_server(flow_name)
+        )
+
+    def _get_flow_cloud(self, flow_name: str) -> "Flow":
         if flow_name not in self.flows:
             raise ValueError("Flow is not contained in this Storage")
         flow_location = self.flows[flow_name]
 
         # Use ref attribute if present, defaulting to "master"
         ref = self.ref or "master"
 
-        client = self._get_bitbucket_client()
+        client = self._get_bitbucket_cloud_client()
+
+        try:
+
+            # Fetch SHA-1 of all branches if ref is not alread a SHA-1 hash
+            SHA1_REGEX = r"^[0-9a-f]{40}$"
+            if not re.match(SHA1_REGEX, ref):
+                branches = client.get(
+                    f"repositories/{self.workspace}/{self.repo}/refs/branches"
+                )
+                branch_hashes = {
+                    branch["name"]: branch["target"]["hash"]
+                    for branch in branches["values"]
+                }
+
+                if ref in branch_hashes:
+                    ref = branch_hashes[ref]
+
+            # Fetch SHA-1 of all tags if ref is not alread a SHA-1 hash
+            if not re.match(SHA1_REGEX, ref):
+                tags = client.get(
+                    f"repositories/{self.workspace}/{self.repo}/refs/tags"
+                )
+                tag_hashes = {
+                    tag["name"]: tag["target"]["hash"] for tag in tags["values"]
+                }
+
+                if ref in tag_hashes:
+                    ref = tag_hashes[ref]
+
+            if not re.match(SHA1_REGEX, ref):
+                raise ValueError("ref not found in this Storage")
+
+            contents_url = (
+                f"repositories/{self.workspace}/{self.repo}/src/{ref}/{flow_location}"
+            )
+            self.logger.info(f"Downloading flow from Bitbucket cloud - {contents_url}")
+            contents = client.get(contents_url)
+
+        except RequestsHTTPError as err:
+            if err.response.status_code == 403:
+                self.logger.error(
+                    "Access denied to repository. Please check credentials."
+                )
+                raise
```

Comment:
```suggestion
                err_msg = "Access denied to repository. Please check credentials."
                self.logger.error(err_msg)
                raise RuntimeError(err_msg) from exc
```

I would do something like this instead so the error is displayed with our message. This would also allow you to easily consolidate your log and re-raise and just set the `msg` in your `if/else`

Reply:
Done